pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -443,11 +443,27 @@ private BrokerResponse query(QueryEnvironment.CompiledQuery query, long requestI
         Tracing.ThreadAccountantOps.clear();
         onQueryFinish(requestId);
       }
-      long executionEndTimeNs = System.nanoTime();
-      updatePhaseTimingForTables(tableNames, BrokerQueryPhase.QUERY_EXECUTION,
-          executionEndTimeNs - executionStartTimeNs);
 
       BrokerResponseNativeV2 brokerResponse = new BrokerResponseNativeV2();
+
+      QueryProcessingException processingException = queryResults.getProcessingException();
+      if (processingException != null) {
+        brokerResponse.addException(processingException);
+        QueryErrorCode errorCode = QueryErrorCode.fromErrorCode(processingException.getErrorCode());
+        if (errorCode == QueryErrorCode.EXECUTION_TIMEOUT) {
+          for (String table : tableNames) {
+            _brokerMetrics.addMeteredTableValue(table, BrokerMeter.BROKER_RESPONSES_WITH_TIMEOUTS, 1);
+          }
+          LOGGER.warn("Timed out executing request {}: {}", requestId, query);
+        }
+        requestContext.setErrorCode(errorCode);
+      } else {
+        brokerResponse.setResultTable(queryResults.getResultTable());
+        long executionEndTimeNs = System.nanoTime();
+        updatePhaseTimingForTables(tableNames, BrokerQueryPhase.QUERY_EXECUTION,
```

Comment:
This is a behavior change, where we only track time for success queries

Reply:
I'm not sure about that. 

In the older code, `submitAndReduce` in IICU could either throw an error or finish successfully. In the first case, we catch the exception and return a BrokerResponseNative without calling `updaterPhaseTimingForTables`. So we only call `updatePhaseTimingForTables` when queries finish successfully